src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -395,20 +423,28 @@ func (d *downsamplerAndWriter) writeAggregatedBatch(
 			}
 		}
 
-		samplesAppender, err := appender.SamplesAppender(opts)
+		result, err := appender.SamplesAppender(opts)
 		if err != nil {
-			return err
+			multiErr = multiErr.Add(err)
```

Comment:
This should be:
```go
if err != nil {
  multiErr = multiErr.Add(err)
  continue
}
```

Otherwise will panic on `result.SamplesAppender.Append...` below since the appender will be nil.